### DISCUSSION TITLE
I want to show my commits in contribution graph

### DISCUSSION THREAD

**Discussion by RashiBaranwal** [community/community#57228] [New to GitHub] [answered]
I was facing issues since many days that when write code and commit it from my vscode to github it get successfully posted but was not shown in the contribution graph.After I linked my gmail properly so now its working but the issue is , the commits which I had done before this that I want to be shown in my contribution graph .Is this possible anyhow.

**Comment by dakshdeepHERE** [comment, marked as answer]
**This happened with me too and if your [mail is right]() on your local device and your GitHub then there can be 3 issues/points you have to consider and before this is how you can confirm your commit email 👉🏼 [Link](https://docs.github.com/en/account-and-profile/setting-up-and-managing-your-personal-account-on-github/managing-email-preferences/setting-your-commit-email-address#about-commit-email-addresses):**
- First you might be sending commits to wrong remote repo you can confirm it by using the command **`git remote -v`** it will return the link and you can confirm the repo you are trying to contribute in.
- Second you might have forked a repo and contributing on the forked repo which again won't show on the contribution graph until your forked repo don't get merged into the active branch of the real repo and there are valid reasons behind this like:
  - Collaboration vs. Independent Work: the purpose of counting contributions on a profile is to highlight the collaborative work done directly on a project. Commits in forks are often considered independent work since they are not made directly in the original repository.

  - Maintainer Control: repository maintainers have control over whether to include contributions made in forks in their project's statistics. They can choose to merge pull requests from forks and attribute the contributions to the original repository, but it's their discretion. 
  - Privacy and Security: forks provide a level of privacy and security, as they allow users to experiment, make changes, and test without directly impacting the original repository. Counting commits in forks as contributions on a profile could potentially expose sensitive work that users may not want to publicize.
- Third, if you are committing on the different branch which is not the active branch then also your commit won't be converted into contribution until you them merged with the active branch.

More over you can read this [doc](https://docs.github.com/en/account-and-profile/setting-up-and-managing-your-github-profile/managing-contribution-settings-on-your-profile/why-are-my-contributions-not-showing-up-on-my-profile) too.

**Reply by dakshdeepHERE** [reply]
If you don't understand about branches check this out 👉🏼 [Link](https://docs.github.com/en/pull-requests/collaborating-with-pull-requests/proposing-changes-to-your-work-with-pull-requests/about-branches)

**Reply by RashiBaranwal** [reply]
Ok thanks I got it...

**Reply by dakshdeepHERE** [reply]
> Ok thanks I got it...

Hey if you need further help I'm always open to help. Good luck with your open source journey.

**Reply by RashiBaranwal** [reply]
Thanks....good luck to you too...

**Reply by Nozibul** [reply]
How can I show the last few months of commits in the contribution graph?

**Comment by Nozibul** [comment]
How can I show the last few months of commits in the contribution graph?
